tests/integration/utils.py

```diff
@@ -242,62 +251,88 @@ def assert_content_manifest_schema(response_data):
-            relative_file_path = os.path.relpath(absolute_file_path, start=source_path)
-            # Assert that content of source file is equal to expected
-            with open(absolute_file_path, "rb") as f:
-                if check_content:
-                    # Download expected file
-                    file_url = expected_file_urls[relative_file_path]
-                    expected_file = requests.get(file_url).content
-                    assert f.read() == expected_file
-                else:
-                    assert os.path.isfile(
-                        absolute_file_path
-                    ), f"File {absolute_file_path} does not exist"
-            files.append(relative_file_path)
-    # Assert that there are no missing or extra files
-    assert set(files) == set(list(expected_file_urls))
+    for dir_to_check, archive_url in expected_files.items():
+        # A directory path to check
+        test_path = os.path.join(source_path, dir_to_check)
+        # If there is no link to expected archive, the directory should be empty
+        if not archive_url:
+            assert (
+                len(os.listdir(test_path)) == 0
+            ), f"Directory: {test_path} not found or not empty as expected."
+        else:
+            dir_identifier = dir_to_check.replace("/", "_")
+            # A directory path with extracted deps
+            deps_data_path = os.path.join(tmpdir, f"test_source_{dir_identifier}")
+            # An archive path with expected data
+            expected_archive = os.path.join(tmpdir, f"archive_{dir_identifier}.tar.gz")
+            # A directory path with extracted expected data
+            expected_data_path = os.path.join(tmpdir, f"expected_data_{dir_identifier}")
+            # Dependencies instead of source code are saved as archives.
+            # Therefore we should extract it firstly
+            if os.path.isdir(test_path):
+                package_root_dir = test_path
+            else:
+                archive_path = test_path
+                shutil.unpack_archive(archive_path, deps_data_path)
+                # deps_data_path is unique and contains only one expected package
+                package_root_dir = os.path.join(deps_data_path, os.listdir(deps_data_path)[0])
+            download_archive(archive_url, expected_archive)
+            shutil.unpack_archive(expected_archive, expected_data_path)
+            # Root directory for expected data of package or dependency
+            expected_package_root_dir = os.path.join(
+                expected_data_path, os.listdir(expected_data_path)[0]
+            )
+            assert os.path.isdir(
+                expected_package_root_dir
+            ), f"Wrong directory path {expected_package_root_dir}."
+            # Compare and assert files in directory with expected data
+            dc = filecmp.dircmp(package_root_dir, expected_package_root_dir)
+            assert len(dc.diff_files) == 0, (
```

Comment:
Looking at the [dircmp documentation](https://docs.python.org/3/library/filecmp.html#the-dircmp-class), I don't think this check is enough. 

```
left_only
    Files and subdirectories only in a.
right_only
    Files and subdirectories only in b.
diff_files
    Files which are in both a and b, whose contents differ according to the class’s file comparison operator.
```

You need to check (at least) all of the above, and you also need to do it recursively for all `subdirs`

Reply:
Ok, I wrote rec function. Sorry for silly realisation.